packages/integrations/useIDBKeyval/index.ts

```diff
@@ -0,0 +1,87 @@
+import type { ConfigurableFlush, MaybeComputedRef, RemovableRef } from '@vueuse/shared'
+import { resolveUnref } from '@vueuse/shared'
+import type { Ref } from 'vue-demi'
+import { ref, shallowRef, watch } from 'vue-demi'
+import { del, get, set, update } from 'idb-keyval'
+
+export interface UseIDBOptions extends ConfigurableFlush {
+  /**
+   * Watch for deep changes
+   *
+   * @default true
+   */
+  deep?: boolean
+
+  /**
+   * On error callback
+   *
+   * Default log error to `console.error`
+   */
+  onError?: (error: unknown) => void
+
+  /**
+   * Use shallow ref as reference
+   *
+   * @default false
+   */
+  shallow?: boolean
```

Comment:
Can we remove this option and the associated code by having the user pass in a computed, ref, or shallowRef?